File: services/server/internal/models/list.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// List represents a task list in the Hot Cross Buns application.
type List struct {
	ID        uuid.UUID  `json:"id"`
	UserID    uuid.UUID  `json:"userId"`
	Name      string     `json:"name" validate:"required,max=255"`
	Color     *string    `json:"color" validate:"omitempty,hexcolor"`
	SortOrder int        `json:"sortOrder"`
	IsInbox   bool       `json:"isInbox"`
	AreaID    *uuid.UUID `json:"areaId"`
	CreatedAt time.Time  `json:"createdAt"`
	UpdatedAt time.Time  `json:"updatedAt"`
	DeletedAt *time.Time `json:"deletedAt"`
}

File: apps/desktop/src-tauri/src/models/list.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct List {
    pub id: String,
    pub name: String,
    pub color: Option<String>,
    pub sort_order: i32,
    pub is_inbox: bool,
    pub area_id: Option<String>,
    pub description: Option<String>,
    pub created_at: String,
    pub updated_at: String,
    pub deleted_at: Option<String>,
}
